src/main/java/org/sonarsource/sonarlint/ls/folders/WorkspaceFolderBranchManager.java

```diff
@@ -80,7 +79,7 @@ public void didBranchNameChange(URI folderUri, @Nullable String branchName) {
         var repo = GitUtils.getRepositoryForDir(Paths.get(folderUri));
         if (repo != null) {
           try (repo) {
-            electedBranchName = GitUtils.electBestMatchingServerBranchForCurrentHead(repo, serverBranchNames, serverBranches.getMainBranchName().orElse(null));
+            electedBranchName = GitUtils.electBestMatchingServerBranchForCurrentHead(repo, serverBranchNames, serverBranches.getMainBranchName());
```

Comment:
```suggestion
            electedBranchName = GitUtils.electBestMatchingServerBranchForCurrentHead(repo, serverBranchNames, serverBranches.getMainBranchName().orElse("master"));
```

Reply:
It does not return an option but a String. And you're right, as null can be returned we probably need to set a default value to master

Reply:
I'm not sure null can be returned here. Getter isn't marked as CheckForNull, and logically server should default to master on it's side.

Reply:
you're right Kirill, I checked and `serverBranches.getMainBranchName()` should always return something

Reply:
On the other hand, `electBestMatchingServerBranchForCurrentHead` can return null. A few lines below we notify the client with the resolved branch. Should we make sure to pass "master" in this case if the elected branch is null ?